### PR TITLE
Run the build integ test in parallel

### PR DESCRIPTION
**Pull request by alpar-t**
Because the randomized runner lives in `buildSrc`, we run these tests with
the Gradle runner, and had no parallelism configured so far.

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -232,6 +232,7 @@ if (project != rootProject) {
     if (isLuceneSnapshot) {
       systemProperty 'test.lucene-snapshot-revision', isLuceneSnapshot[0][1]
     }
+    maxParallelForks System.getProperty('tests.jvms', project.rootProject.ext.defaultParallel) as Integer
```
Comment (mark-vieira) [inline buildSrc/build.gradle]: `BuildPlugin.findDefaultParallel()` will return the string `auto` on non-unix systems, which works for `RandomizedRunner` but will blow up when we try to cast it to an `Integer` here. We need a different strategy here. Perhaps just falling back to `Runtime.availableProcessors()` if `defaultParallel` returns `auto`.

**Review by mark-vieira — CHANGES_REQUESTED** [review]
One comment to address as I don't think this will work on Windows as-is.

File: buildSrc/build.gradle (hunk 1 above)
Comment (atorok) [reply buildSrc/build.gradle]: good catch! thanks. I added a check for "auto" and capped available processors to 4. This is the same as what "auto" means for randomized testing.

**Review by mark-vieira — APPROVED** [review]
Changes LGTM 👍

**Pull request merged**